src/Gui/Selection/Selection.cpp

```diff
@@ -420,6 +422,15 @@ void SelectionSingleton::notify(SelectionChanges &&Chng)
             notify = true;
         }
         if(notify) {
+            // Notify the view provider of the object as well.
+            if(const auto* doc = App::GetApplication().getDocument(msg.pDocName)) {
+                if(const auto* obj = doc->getObject(msg.pObjectName)) {
+                    if (auto* vp = Application::Instance->getViewProvider(obj)) {
+                        vp->onSelectionChanged(msg);
+                    }
+                }
+            }
+

```

Comment:
Can't we change it to early exit? These are 4 ifs nested, that is quite a lot

Reply:
I'm not sure what to do here. While I agree that early-exists are nice, we do run code afterwards.
To make it work we would need to create a lambda function for it.

An implementation would looks something like this:
```suggestion
            const auto notifyObjectViewProvider = [](const auto& msg) {
                const auto* doc = App::GetApplication().getDocument(msg.pDocName);
                if (!doc) return;
    
                const auto* obj = doc->getObject(msg.pObjectName);
                if (!obj) return;
    
                auto* vp = Application::Instance->getViewProvider(obj);
                if (!vp) return;
    
                vp->onSelectionChanged(msg);
            };

            notifyObjectViewProvider(msg);

```

It gets quite big by doing this change, I don't mind tritao's original version but I'm open for either.

Reply:
Yeah, I'm fine with that 
```suggestion
            // Notify the view provider of the object as well.
            if (const auto* doc = App::GetApplication().getDocument(msg.pDocName)) {
                if (const auto* obj = doc->getObject(msg.pObjectName)) {
                    if (auto* vp = Application::Instance->getViewProvider(obj)) {
                        vp->onSelectionChanged(msg);
                    }
                }
            }


```

Reply:
Instead of a static function, you can use a normal function in an anonymous namespace.

```
namespace {
    void foo() {
        // ...
    }
}
```

This is basically the same thing as a static function, but the c++onic way, instead of the ye olde c-way.